main.py

```diff
@@ -1,12 +1,251 @@
+                video.get_video(title=title)
+            elif select_by=="id":
+                id = input("Which video id would you like to select? \n")
+                if id.isnumeric():
+                    id = int(id)
+                    video.get_video(id=id)
+            else:
+                print("Could not locate. Please enter a valid id or title we have in our collection.")
+            
+            if video.selected_video:
+                print()
+                print("Selected Video: ", video.selected_video)
+        # "6": "New Customer Sign Up"    
+        elif choice == '6':
+            print("Let's get started on adding a new customer!\n ")
+            name=input("What is the name of the customer? \n").title()
+            postal_code =input("What's the customer's postal code? \n")
+            phone =input("What is the customer's phone number? \n")
+            response = customer.create_new_customer(name=name, postal_code =postal_code, phone=phone)
+            print()
+            print("New Customer added to the Retro Video Store:", name.title())
+        ## "7": "Update a Customer's Information"    
+        elif choice == '7':
+            print(f"Let's update the customer's information: {customer.selected_customer}")
+            name=input("What updates do you want to make to the customer's name?\n")
+            postal_code=input("What updates do you want to make to the customer's postal code?\n")
+            phone = input("What updates do you want to make to the customer's phone?\n")
+            response = customer.update_customer(name=name, postal_code=postal_code, phone=phone)
+            print()
+            print("Updated Customer:", response)
+        ## "8": "Delete 1 Customer "    
+        elif choice == '8':
+            customer.delete_customer()
+            print()
+            print("The Customer has been deleted.")
+            print()
+            print(customer.list_customers())
+        ## "9": "View  Info on 1 Customer"    
+        elif choice == '9':
+            select_by = input("Would you like to select by? Enter name or id: \n ")
+            if select_by=="name":
+                name = input("Which customer name would you like to locate? \n")
+                customer.get_customer(name=name)
+            elif select_by=="id":
+                id = input("Which customer id would you like to locate? \n")
+                if id.isnumeric():
+                    id = int(id)
+                    customer.get_customer(id=id)
+            else:
+                print("Could not locate. Please enter a valid customer id or name")
+            
+            if customer.selected_customer:
+                print()
+                print("Selected Customer: ", customer.selected_customer)
+        ## "10": "View all Customers"    
+        elif choice == '10':
+            for customer in customer.list_customers():
+                print(customer)
```

Comment:
In this loop we accidently override the `customer` instance with a customer dictionary that is returned in the json from `list_customers`. This raises an exception later on when we try to run the line `customer.print_selected`. Consider naming your customer instance something like `customer_manager` since it provides all the methods to handle the customer routes.